### DISCUSSION TITLE
Multi-instance docs + sidebar

### DISCUSSION THREAD

**Discussion by marikaner** [facebook/docusaurus#7552] [Q&A] [answered]
I have multi-instance docs with two products. Each of the products have their own sidebar.
I have two navbar items, that point to some product-specific docs, one for each of the products.
I don't want to repeat the navbar item in the sidebar. However, I want to keep the product related sidebar open, when the navbar item was clicked.

Normally, there is `displayed_sidebar` for this. But I struggle to point to the correct id with the multi-instance plugin. All my sidebars are called "sidebar", but this does not seem to be recognized. Error: `Doc with ID about wants to display sidebar sidebar but a sidebar with this name doesn't exist`.

Any ideas how to combine those?

**Comment by slorber** [comment]
> I don't want to repeat the navbar item in the sidebar. 

I don't know what you mean here

> However, I want to keep the product related sidebar open, when the navbar item was clicked.

Isn't this already the case?

> Normally, there is displayed_sidebar for this. But I struggle to point to the correct id with the multi-instance plugin. All my sidebars are called "sidebar", but this does not seem to be recognized. 

This only works within a docs plugin instance, not across multiple instances. 

That's why 2 sidebars sharing the same name do not conflict with each others somehow: they are isolated by their docs instance.

---

> Any ideas how to combine those?

I believe you are looking for the `docsPluginId` attribute that is available on multiple docs-related navbar items.

https://docusaurus.io/docs/api/themes/configuration#navbar-doc-sidebar-docsPluginId

For example:

```js
module.exports = {
  themeConfig: {
    navbar: {
      items: [
        {
          type: 'docSidebar',
          sidebarId: 'sidebar',
          label: 'Product 1',
          docsPluginId: "product-1",
        },
        {
          type: 'docSidebar',
          sidebarId: 'sidebar',
          label: 'Product 2',
          docsPluginId: "product-2",
        },
      ],
    },
  },
};
```

---

It's impossible to help more without a repro (use docusaurus.new 🙏 ). I don't even know which kind of sidebar item you are using.

**Comment by marikaner** [comment, marked as answer]
TLDR; It just works with `displayed_sidebar: sidebar`. I can't even reproduce the issue anymore.

I am leaving this here for reference, in case someone is trying to achieve something similar.
I created a [repro example](https://stackblitz.com/edit/github-v75b7y?file=docusaurus.config.js).
In this example you will find 4 navbar items:
- left: Prod 1, Prod 2
- right API Docs (Prod 1), API Docs (Prod 2)

For the nav bar item "Prod 1" the sidebar has 2 doc links (overview + api-docs). The nav bar item "API Docs (Prod 1)" works as I want it to work, while staying with the "Prod 1" sidebar. Now, I just want to get rid of the api-docs link in the sidebar. 

I tried to achieve this for `prod2`: I first removed the api-docs from the sidebar, but then the sidebar completely disappears when navigating to the nav bar item "API Docs (Prod 2)".
From [this documentation](https://docusaurus.io/docs/sidebar/multiple-sidebars) it reads like for this, I could use `displayed_sidebar`. And well - it works. I am not sure what I was doing wrong before.

**Reply by slorber** [reply]
So what I understand: you just want to display a document that is not in a sidebar, but still display the sidebar for that doc.

Yes: `displayed_sidebar` should work for that use-case

**Reply by marikaner** [reply]
Yes, correct. Sorry for the confusion, I guess this was just a typo or similar...
